src/llmcompressor/recipe/recipe.py

```diff
@@ -593,13 +593,20 @@ def _load_json_or_yaml_string(content: str) -> Dict[str, Any]:
     # try loading as json first, then yaml
     # if both fail, raise a ValueError
     try:
-        return json.loads(content)
+        ret = json.loads(content)
     except json.JSONDecodeError:
         try:
-            return yaml.safe_load(content)
+            ret = yaml.safe_load(content)
         except yaml.YAMLError as err:
             raise ValueError(f"Could not parse recipe from string {content}") from err
 
+    if not isinstance(ret, dict):
```

Comment:
Would this value error be redundant? In what cases would you not get a dictionary and not have hit the Value Error on line 601?

Reply:
@dsikka This happens for any string which does not look like a dictionary, for example
```python3
import yaml
assert isinstance(yaml.safe_load("1"), int)
assert isinstance(yaml.safe_load("asdf"), str)
assert isinstance(yaml.safe_load("my/mistaken/path"), str)
```

Reply:
interesting, so yaml.safe_load() will load the strings without issue?

Reply:
Yes, `safe` refers to how pyyaml will throw an error if the string contains dangerous python objects such as functions, libraries, ect. It does not guarantee that the deserialized value is of dictionary type.